### PR TITLE
add a new command `runCommands` that allows to run several commands sequentially

### PR DESCRIPTION
**Pull request by ulugbekna**

### CODE REVIEW

File: src/vs/workbench/contrib/commands/common/commands.contribution.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		}
+	}
+
+	private _runCommand(commandService: ICommandService, cmd: RunnableCommand) {
+		let commandID: string, commandArgs;
+
+		if (typeof cmd === 'string') {
+			commandID = cmd;
+		} else {
+			commandID = cmd.id;
```
Comment (aeschli) [inline src/vs/workbench/contrib/commands/common/commands.contribution.ts:46]: Better check that cmd.id is a string

**Review by aeschli — CHANGES_REQUESTED** [review]
There's a way to add a description to the command as well as a schema:
https://github.com/microsoft/vscode/blob/9ca779d59799d0aa335e771d413519c09e579256/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts#L287

**Comment by ulugbekna** [comment]
> Here's another description (for editorScroll)

Are you talking about validation of `keybindings.json`? If yes, let's move to this [issue](https://github.com/microsoft/vscode/issues/176629) for further discussion? I don't understand why that schema is used for `keybindings.json`

**Comment by ulugbekna** [comment]
Thanks! Addressed the review comments.

**Review by aeschli — APPROVED** [review]

**Comment by aeschli** [comment]
Looks good.
It would be cool to get code completions om the command names and validations on the args.
The current schema knows has all the rules to validate command. It could be restructured to make these rules to be referred from your schema.
Let me know it you want me to go into more details.

**Comment by ulugbekna** [comment]
> It would be cool to get code completions om the command names and validations on the args.
> The current schema knows has all the rules to validate command. It could be restructured to make these rules to be referred from your schema.
> Let me know it you want me to go into more details.

Yes, I would absolutely love to have completion & validation. 

1. For completion, I need to update the configuration-editing extension, right? 
2. For validation, yes, I would love more details on how that could look like

File: src/vs/workbench/contrib/commands/common/commands.contribution.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as nls from 'vs/nls';
+import { Action2, registerAction2 } from 'vs/platform/actions/common/actions';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+import { ServicesAccessor } from 'vs/platform/instantiation/common/instantiation';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+
+type RunnableCommand = string | { id: string; args: any[] };
```
Comment (YKG) [inline src/vs/workbench/contrib/commands/common/commands.contribution.ts:12]: should we replace `id` with `command`

**Comment by randseay** [comment]
As a user I'm really excited for this feature! Thank you 👏

**Pull request merged**